### DISCUSSION TITLE
SO 2.3.21 slow starting on ESXi

### DISCUSSION THREAD

**Discussion by gbrodt** [Security-Onion-Solutions/securityonion#2991] [Unsupported Versions] [answered]
Hi,

In my test pod for the SO 2.3.21 seems to be starting very slow.  These are stand alone with 17GB RAM 4 cores and 200GB running on a esxi 6.5. The host server does not seem to be under load. When doing a sudo so-status the machine goes to a waiting state.  I've seen times of  about an hour before it's "ok". Any thoughts or suggestions why this may be happening? Once the status is OK all seems to run good.

**Comment by petiepooo** [comment]
You're likely just very short of resources.  Get a shell on the system after startup and run top or htop to see how busy your CPUs are and how much memory is consumed..  Pay attention to the IO wait state percentage ("wa" in top).  Or run 'vmstat 2 30' to see how swappy things are (SI & SO columns).

Even on a lightly loaded system (but with 7 zeek workers and 7 suricata threads), I'm showing 40GB RAM in use, not including the page cache.  Elasticsearch by itself is consuming 26GB resident.

**Comment by innovate-support** [comment]
The resources you specified are low for a StandAlone box VM.  I'd double the Mem and Cores.  I've tried to run StandAlone VM's for years now with similar specs and they don't run well at all.  They're slow and periodically just lock up completely and require a hard reset of the VM, which is bad for the system.  I've now switched to Distributed environment where the Manager VM is running on box B, and Sensor VM is running on box A where all the other VM's are that I want to monitor.  Getting the Manager off of box A seemed to be the key for me.  I also don't do Full Packet Capture, which means I have Steno disabled. 

I gave my Manager 32G Mem and 8 cores, and my Sensors 16G Mem and 4 cores and they've been running fine for weeks now.  I'm slowly feeding more traffic into them to see if they choke.

**Comment by gbrodt** [comment]
Thank you for the responses and I'll do some checking after I get my current fire taken care of.  The install is for a very minimal setup for students to work with and I was going by the doc for the recommended minimum for a stand alone, guess that's not to functional.

**Reply by TOoSmOotH** [reply]
Usually slow starts like that are due to poor disk performance. A standalone install is doing quite a few things especially with everything turned on. You have 2 ES instances, influxdb, and MySQL battling it out for use of the disk. I woudl watch the io performance on boot and see if that is the cause.

**Comment by gbrodt** [comment, marked as answer]
So I've checked the IO and wait states and they seem to be fine. If I recall correctly I did a change of addressing (no no I know) as these were being used for student pods. In the end I did a fresh install and bumped memory up a little (24G) as they only have 1 process running for zeek and suricata. They seem to be starting fine and running well on the esx server now.
